lib/common/components/Login.js

```diff
@@ -1,31 +1,35 @@
 // @flow
 
-import {Component} from 'react'
+import { useAuth0 } from '@auth0/auth0-react'
+// $FlowFixMe useEffect not recognized by flow.
+import { useEffect } from 'react'
+import { browserHistory } from 'react-router'
 
-import * as userActions from '../../manager/actions/user'
-import auth0 from '../user/Auth0Manager'
-
-import type {Props as ContainerProps} from '../containers/Login'
-
-type Props = ContainerProps & {
-  onHide: () => void,
-  push: string => void,
-  receiveTokenAndProfile: typeof userActions.receiveTokenAndProfile,
-  redirectOnSuccess: string
+type Props = {
+  redirectOnSuccess?: string
 }
 
-export default class Auth0 extends Component<Props> {
-  componentDidMount () {
-    // Show lock when component mounts. NOTE: render method returns null.
-    auth0.loginWithLock(this.props)
-  }
-
-  componentWillUnmount () {
-    // Hide lock when component unmounts
-    auth0.hideLock()
-  }
+export default function Login ({
+  redirectOnSuccess = window.location.path || '/home'
+}: Props) {
+  const auth0 = useAuth0()
+  const { isAuthenticated, isLoading } = auth0
+  useEffect(() => {
+    if (!isLoading && redirectOnSuccess) {
+      if (!isAuthenticated) {
+        // Show the login page if the user state is not logged in.
+        auth0.loginWithRedirect({
+          appState: {
+            returnTo: redirectOnSuccess
```

Comment:
Is this just the regular `auth0` library? I don't think this is the correct argument structure for the redirect url. Perhaps that is why I'm having trouble logging in?
https://auth0.com/docs/libraries/auth0-single-page-app-sdk#login-with-redirect

Reply:
I changed it to redirect_uri as specified in the documentation above, which results in a more helpful auth0 error. 
<img width="653" alt="image" src="https://user-images.githubusercontent.com/92878779/187320834-2863f1a1-4bd8-43af-8ba9-7f0fe1763e40.png">
Adding `http://localhost:9966` to the uri causes the error message to change to this. `unauthorized_client: Callback URL mismatch. http://localhost:9966/home/ is not in the list of allowed callback URLs`

Reply:
The scenario to keep in mind involves a user pasting a URL to a feed version or editor page (e.g. URLs containing `/feed/...`, `/editor/...`). When directly using `redirectUri`, each of the full paths must be whitelisted as callback URLs, which is impractical, whereas with `appState.returnTo`, only the root path needs to be passed to `redirectUri` in the `<Auth0Provider>` and whitelisted in the callback URLs.